liquidity-rewards/contracts/UniswapPoolRewards.sol

```diff
@@ -51,6 +52,12 @@ contract UniswapPoolRewards is PoolTokenWrapper {
     constructor(address elimuToken_, address poolToken_) {
         elimuToken = IERC20(elimuToken_);
         poolToken = IERC20(poolToken_);
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    function setRewardRatePerSecond(uint256 rewardRatePerSecond_) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender));
```

Comment:
Why not just use the `onlyRole(DEFAULT_ADMIN_ROLE)` modifier?

Reply:
@hhio618 Thanks for the heads up, since I wasn't aware if that modifier 🙂

Do you mean like this?: https://github.com/elimu-ai/web3-smart-contracts/pull/67/commits/021dc7b652575ff9a93740a8aea1b6c3e0a8fef5

Reply:
You're welcome :)